app/routes/synonyms.$synonymId._index.tsx

```diff
@@ -0,0 +1,176 @@
+  const intent = getFormDataString(data, 'intent')
+  const synonymId = getFormDataString(data, 'synonymId')
+  if (!synonymId) return redirect('/synonyms')
+  if (intent === 'edit') {
+    const additions =
+      getFormDataString(data, 'addSynonyms')?.split(',') || ([] as string[])
+    const filtered_additions = additions.filter((add) => add)
+    const subtractions =
+      getFormDataString(data, 'deleteSynonyms')?.split(',') || ([] as string[])
+    const filtered_subtractions = subtractions.filter((sub) => sub)
+    await putSynonyms({
+      synonymId,
+      additions: filtered_additions,
+      subtractions: filtered_subtractions,
+    })
+
+    return redirect(`/synonyms/${synonymId}`)
+  }
+  if (intent === 'delete') {
+    await deleteSynonyms(synonymId)
+
+    return redirect('/synonyms')
+  }
+  return null
+}
+
+export default function () {
+  const { synonymId, eventIds } = useLoaderData<typeof loader>()
+  const [deleteSynonyms, setDeleteSynonyms] = useState([] as string[])
+  const [synonyms, setSynonyms] = useState(eventIds || [])
+  const [addSynonyms, setAddSynonyms] = useState([] as string[])
+  const [newSynonym, setNewSynonym] = useState('')
+
+  return (
+    <>
+      <ButtonGroup>
+        <h1>Synonym Group</h1>
+        <ButtonGroup className="margin-bottom-2">
+          <Link to="/synonyms" className="usa-button">
+            <div className="display-inline">
+              <Icon.ArrowBack
+                role="presentation"
+                className="position-relative"
+              />
+            </div>
+          </Link>
+          <Form method="POST">
+            <input type="hidden" name="intent" value="delete" />
+            <input
+              type="hidden"
+              name="synonymId"
+              value={synonymId || undefined}
+            />
+            <Button type="submit">
+              <Icon.Delete
+                role="presentation"
+                className="bottom-aligned margin-right-05"
+              />
+            </Button>
+          </Form>
+        </ButtonGroup>
+      </ButtonGroup>
+      <p>
+        Synonym groupings are limited to 25 synonymous event identifiers. If you
+        are adding an event identifier that is already part of a group, it will
+        be removed from the previous association and added to this group.
+      </p>
+      <Form>
+        <FormGroup>
+          <ButtonGroup>
+            <input
+              placeholder="event id"
+              value={newSynonym}
+              key="synonym"
+              onChange={(e) => {
+                setNewSynonym(e.currentTarget.value)
+              }}
+            />
+            <Button
+              type="button"
+              onClick={(e) => {
+                if (!newSynonym) return
+                setSynonyms([...synonyms, newSynonym])
+                setAddSynonyms([...addSynonyms, newSynonym])
+                setNewSynonym('')
+              }}
+            >
+              <Icon.Add /> Add
```

Comment:
Set `alt=""` or `role="presentation"` to indicate that this icon is not relevant for screen readers.